### ISSUE TITLE
Not reporting amps (or watts)

### ISSUE THREAD

**Issue by wabmester** [alexocn/m365_bms_monitor#2]
@alexocn thanks for making this.   The version you checked in just spoofs the app.   Did you make one that actually passes
the messages on to the BMS?

**Comment by alexocn** [comment]
What do you mean?
It actually sends  requests and receives the BMS info and then passes it to the phone app. 
If you don't have BMS connected there are commented out hardcoded examples to see if your app works, but by default it doesn't do "spoofing" but forwarding instead. 
Not sure why it doesn't report amp, those should be the part of the 0x31 message which are processed.  I am not sure I actually check them myself since I used the current measurement on the load which should be more accurate.

**Comment by wabmester** [comment]
I suppose it's possible that the serial packets just aren't getting to my BMS, but I'm not seeing the serial communication in your code.   I'm only getting the spoofed messages on the app that are being sent from the code starting at line 68 (or line 104 when I define NBOTFORMAT).

The only use of the hardware UART I see is in the form of debug messages (which I get over the serial monitor).

**Comment by alexocn** [comment]
You are right, my bad. I didn't check in the "bmsmonitor" itself, only "emulator" which I used for testing of the APP.
Fixed, see  m365_esp32BLE_bmsmonitor.ino  file, this is the main file. 
//I am using Serial2_TX(17pin)for transmit to BMS,  Serial0_RX(regular) to receive, while Serial0_TX is used for debugging to USB UART.

**Comment by wabmester** [comment]
Works great!    Got it working with my M5StickC.   Was hoping it'd work w/o a pull-up, but no such luck
![20220414_205825](https://user-images.githubusercontent.com/39422301/163516276-bb801931-0d92-4f82-9c1e-dbfdbcacee81.jpg)
.

**Comment by alexocn** [comment]
I am glad it worked out for you. I am guessing you are the only one who replicated the project so far. 
I tried multiple boards (Uno,Nano,Mega, ESP8266,and couple of different ESP32). For most pull-up resistor on Arduino/ESP RX is needed. I think only Nano and I think ESP8266(D1mini) worked without pull-up.  Transmit Arduino/ESP to BMS seems a lot more robust and worked even with SoftwareSerial(which didn't work for me on RX side in any configuration).
